sdks/unity/AgonesSdk.cs

```diff
@@ -0,0 +1,239 @@
+        /// <param name="key">annotation key</param>
+        /// <param name="value">annotation value</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// The task result contains that the request is success or failure.
+        /// </returns>
+        public async Task<bool> SetAnnotation(string key, string value)
+        {
+            string json = JsonUtility.ToJson(new KeyValueMessage(key, value));
+            return await SendRequestAsync("/metadata/annotation", json, UnityWebRequest.kHttpVerbPUT);
+        }
+        #endregion
+
+        #region AgonesRestClient Private Methods
+        void Health()
+        {
+            _ = SendRequestAsync("/health", "{}");
+        }
+
+        async Task<bool> SendRequestAsync(string api, string json, string method = UnityWebRequest.kHttpVerbPOST)
+        {
+            // To prevent that an async method leaks after destroying this gameObject
+            cancellationTokenSource.Token.ThrowIfCancellationRequested();
+
+            var req = new UnityWebRequest(sidecarAddress + api, method)
+            {
+                uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(json)),
+                downloadHandler = new DownloadHandlerBuffer()
+            };
+            req.SetRequestHeader("Content-Type", "application/json");
+
+            await new AgonesAsyncOperationWrapper(req.SendWebRequest());
+
+            bool ok = req.responseCode == 200;
+
+            if (ok)
+            {
+                Log($"Agones SendRequest ok: {api} {req.downloadHandler.text}");
+            }
+            else
+            {
+                Log($"Agones SendRequest failed: {api} {req.error}");
+            }
+
+            return ok;
+        }
+
+        void Log(object message)
+        {
+            if (!logEnabled) { return; }
+
+#if UNITY_EDITOR
+            Debug.Log(message);
+#else
+            Console.WriteLine(message);
+#endif
+        }
+        #endregion
+
+        #region AgonesRestClient Nested Classes
+        class AgonesAsyncOperationWrapper
+        {
+            public UnityWebRequestAsyncOperation AsyncOp { get; }
+            public AgonesAsyncOperationWrapper(UnityWebRequestAsyncOperation unityOp)
+            {
+                AsyncOp = unityOp;
+            }
+
+            public AgonesAsyncOperationAwaiter GetAwaiter()
+            {
+                return new AgonesAsyncOperationAwaiter(this);
+            }
+        }
+
+        class AgonesAsyncOperationAwaiter : INotifyCompletion
+        {
+            private UnityWebRequestAsyncOperation asyncOp;
+            private Action continuation;
+
+            public AgonesAsyncOperationAwaiter(AgonesAsyncOperationWrapper wrapper)
+            {
+                asyncOp = wrapper.AsyncOp;
+                asyncOp.completed += OnRequestCompleted;
+            }
+
+            public bool IsCompleted => asyncOp.isDone;
+
+            public void GetResult()

```

Comment:
This method's name implies that it returns the result of the operation, when it actually just seems to unsubscribe the completed callback and unset its fields. Please either change the name or have the function return an expected value, in addition to adding `<summary>` descriptions.

Reply:
C# Awaiter Pattern requires that the return type of the `GetAwaiter` method has `GetResult()`,
and `AgonesAsyncOperationAwaiter` does not return a value in this case.

https://github.com/googleforgames/agones/pull/818/files#diff-f35409fa73f6901e4704a2951810b018R159

So I can not change this function name and do not need to return a value.

Reply:
Makes sense, just please add a method comment explaining such. Thanks